placement/algo/helper.go

```diff
@@ -38,9 +38,9 @@ type PlacementHelper interface {
 	// MoveOneShard moves one shard between 2 hosts
 	MoveOneShard(from, to placement.HostShards) bool
 	// MoveShard moves a particular shard between 2 hosts
-	MoveShard(shard uint32, from, to placement.HostShards, rackCheck bool) bool
-	// CanMoveShardToRack checks if it's valid to move a shard to target rack
-	CanMoveShardToRack(shard uint32, from placement.HostShards, toRack string) bool
+	MoveShard(shard uint32, from, to placement.HostShards) bool
+	// HasNoRackConflict checks if it's valid to move a shard to target rack
```

Comment:
`checks if the rack constraint is violated if the given shard is moved to the target rack`